Demo/Demo/Models/CreateOrderParams.swift

```diff
@@ -2,6 +2,21 @@ struct CreateOrderParams: Codable {
 
     let intent: String
     var purchaseUnits: [PurchaseUnit]?
+    var paymentSource: VaultCardPaymentSource?
+
+    init(intent: String, purchaseUnits: [PurchaseUnit]?, isVaulting: Bool = false, customerID: String? = nil) {
+        self.intent = intent
+        self.purchaseUnits = purchaseUnits
+        if isVaulting {
+            var customer: CardVaultCustomer?
+            if let customerID {
+                customer = CardVaultCustomer(id: customerID)
+            }
+            let attributes = Attributes(vault: Vault(storeInVault: "ON_SUCCESS"), customer: customer)
+            let card = VaultCard(attributes: attributes)
+            self.paymentSource = VaultCardPaymentSource(card: card)
+        }
+    }
```

Comment:
Should we move this logic out of the `init`? It seems like we could do this check at a later point or in the view directly vs the init of the model.

Reply:
Not sure what you mean Jax. The vaulting options are passed in the request body, so I just passed the params into here.

Reply:
Not sure what you mean Jax. We pass in vault params in the request body so I am passing in the vault options into param that we use for order creation in demo app.

Reply:
Typically we want our models to be "dumb" - if you look at the rest of the models in this directory, the `init` is typically super simple and logic/mutations are handled elsewhere. We should be able to do something like this...

In `CreateOrderParams`:
```swift
struct CreateOrderParams: Codable {

    let intent: String
    let isVaulting: Bool
    var purchaseUnits: [PurchaseUnit]?
    var paymentSource: VaultCardPaymentSource?

    init(intent: String, purchaseUnits: [PurchaseUnit]?, isVaulting: Bool = false, customerID: String? = nil) {
        self.intent = intent
        self.purchaseUnits = purchaseUnits
        self.isVaulting = isVaulting
    }
}
```

In `CardPaymentViewModel`:
```swift
    func createOrder(
        amount: String,
        selectedMerchantIntegration: MerchantIntegration,
        intent: String,
        isVaulting: Bool,
        customerID: String? = nil
    ) async throws {
...
        var orderRequestParams = CreateOrderParams(
            intent: intent,
            purchaseUnits: [PurchaseUnit(amount: amountRequest)],
            isVaulting: isVaulting,
            customerID: customerID
        )

        if isVaulting {
            var customer: CardVaultCustomer?
            if let customerID {
                customer = CardVaultCustomer(id: customerID)
            }
            let attributes = Attributes(vault: Vault(storeInVault: "ON_SUCCESS"), customer: customer)
            let card = VaultCard(attributes: attributes)
            orderRequestParams.paymentSource = VaultCardPaymentSource(card: card)
        }

        do {
...
```
^ this is just pseudo code so worth adding some breakpoints to test it's working as expected.

Reply:
Thank you. I will take a look.

Reply:
Oh I see, you mean,  attach the paymentSource outside of the CreateOrderParams struct, in the viewModel's createOrder function.

Reply:
We could potentially have other options if requirements come in for these parameters. So should the logic be 
in the CreateOrderParams? or out in createOrder function? It could be header or it could be included in the purchase Units.

Reply:
I think it's less about creating something outside of the struct and more about separation of concerns. The model `CreateOrderParams ` shouldn't need to know about unrelated types, it should only know about itself. I think we could even take it a step further (same as before - you'll want to confirm everything looks as expected) by doing something like this:
```swift
struct CreateOrderParams: Codable {

    let intent: String
    let purchaseUnits: [PurchaseUnit]?
    let paymentSource: VaultCardPaymentSource?
    let shouldVault: Bool

    init(
        intent: String,
        purchaseUnits: [PurchaseUnit]?,
        paymentSource: VaultCardPaymentSource? = nil,
        shouldVault: Bool = false
    ) {
        self.intent = intent
        self.purchaseUnits = purchaseUnits
        self.paymentSource = paymentSource
        self.shouldVault = shouldVault
    }
}
```

```swift
class CardPaymentViewModel: ObservableObject, CardDelegate {

    ...

    func createOrder(
        amount: String,
        selectedMerchantIntegration: MerchantIntegration,
        intent: String,
        shouldVault: Bool,
        customerID: String? = nil
    ) async throws {
        // might need to pass in payee as payee object or as auth header

        var customer: CardVaultCustomer?
        if shouldVault, let customerID {
            customer = CardVaultCustomer(id: customerID)
        }

        let amountRequest = Amount(currencyCode: "USD", value: amount)
        let attributes = Attributes(vault: Vault(storeInVault: "ON_SUCCESS"), customer: customer)
        let card = VaultCard(attributes: attributes)

        // TODO: might need to pass in payee as payee object or as auth header
        let orderRequestParams = CreateOrderParams(
            intent: intent,
            purchaseUnits: [PurchaseUnit(amount: amountRequest)],
            paymentSource: VaultCardPaymentSource(card: card),
            shouldVault: shouldVault
        )
    ...
}
```
Curious to hear your thoughts! 😄

Reply:
Addressed in da203cbe1bbbd9c4cf91b7153fcee4b3e82b2547

Reply:
I like it! Will use it.

Reply:
Addressed in a3a138e88d058c26a3b39aa711fd244b15569114